app/controllers/transactions_controller.rb

```diff
@@ -57,46 +36,21 @@ def index
 			}
 			@all_transactions.push(resp)
 		end
-
 	end
 
 	def dashboard
-		@total_rev = 0
-		@daily_sales = [0, 0, 0, 0, 0, 0, 0]
-		@hourly_sales = {}
-		@movie_sales = []
-
-		Transaction.all.each do |transaction|
-			# Add the total from the transaction to total_rev
-			@total_rev += transaction.cost
-			# Find the showing the transaction belongs to
-			showing = transaction.showing
-			# Add the total from the transaction to showings day of the week
-			@daily_sales[showing.date.wday] += transaction.cost
+		dash_services = DashboardService
```

Comment:
To make this more configurable (what if you want to run the service on a smaller/scoped portion of transactions?), I would pass in the transaction records (e.g. `DashboardService.new(transactions: Transaction.all)` as an argument to initializer